dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/classifier/LabelTree.java

```diff
@@ -0,0 +1,447 @@
+        LabelTreeNode node = getNode(LabelTreeNode.makeBasicNode(labelId));
+
+        if (node == null)
+            return null;
+
+        return node.getLabelName();
+    }
+
+    public boolean setLabelName(String labelId, String labelName) {
+        LabelTreeNode node = getNode(LabelTreeNode.makeBasicNode(labelId));
+
+        if (node == null)
+            return false;
+
+        node.setLabelName(labelName);
+        return true;
+    }
+
+    public String getLabelDescription(String labelId) {
+        LabelTreeNode node = getNode(LabelTreeNode.makeBasicNode(labelId));
+
+        if (node == null)
+            return null;
+
+        return node.getLabelDescription();
+    }
+
+    public boolean setLabelDescription(String labelId, String labelDesc) {
+        LabelTreeNode node = getNode(LabelTreeNode.makeBasicNode(labelId));
+
+        if (node == null)
+            return false;
+
+        node.setLabelDescription(labelDesc);
+        return true;
+    }
+
+    public boolean addEdge(String parent, String child) {
+        LabelTreeNode parentNode = LabelTreeNode.makeBasicNode(parent);
+        LabelTreeNode childNode = LabelTreeNode.makeBasicNode(child);
+
+        return addEdge(parentNode, childNode);
+    }
+
+    public boolean addEdges(String parent, Set<String> children) {
+        boolean success = true;
+
+        for (String child : children) {
+            success = addEdge(parent, child);
+
+            if (success == false)
+                break;
+        }
+
+        return success;
+    }
+
+    public int getDepth(String labelId) {
+        LabelTreeNode node = LabelTreeNode.makeBasicNode(labelId);
+        return getDepth(node);
+    }
+
+    public List<String> getAllParents(String labelId) {
+        List<LabelTreeNode> parentNodes = getAllParents(LabelTreeNode.makeBasicNode(labelId));
+
+        if (parentNodes == null)
+            return null;
+
+        List<String> parents = new ArrayList<>(parentNodes.size());
+
+        for (LabelTreeNode p : parentNodes) {
+            parents.add(p.getLabel());
+        }
+
+        return parents;
+    }
+
+    public List<String> getBreadthOrderedLabelList() {
+        List<LabelTreeNode> nodes = getBreadthOrderedNodeList();
+
+        if (nodes == null)
+            return null;
+
+        List<String> labelIds = new ArrayList<>(nodes.size());
+
+        for (LabelTreeNode p : nodes) {
+            labelIds.add(p.getLabel());
+        }
+
+        return labelIds;
+    }
+
+    public List<String> getDepthOrderedLabelList() {
+        List<LabelTreeNode> nodes = getDepthOrderedNodeList();
+
+        if (nodes == null)
+            return null;
+
+        List<String> labelIds = new ArrayList<>(nodes.size());
+
+        for (LabelTreeNode p : nodes) {
+            labelIds.add(p.getLabel());
+        }
+
+        return labelIds;
+    }
+
+    public boolean isLeaf(String labelId) {
+        LabelTreeNode node = LabelTreeNode.makeBasicNode(labelId);
```

Comment:
The intention was to prevent the end-user from accessing the underlying tree structure, and only allow manipulations using the NodeIDs. Furthermore, this Tree implementation is not a general purpose Tree implementation, and thus, such functions are seldom used.